circuits/src/ecc/hash_to_curve/htc_gadget.rs

```diff
@@ -32,6 +32,12 @@ use crate::{
 };
 
 #[derive(Clone, Debug)]
+#[cfg_attr(
+    feature = "extraction",
+    derive(picus::NoChipArgs, picus::InitFromScratch),
+    from_scratch(H),
+    from_scratch(E)
+)]
```

Comment:
One instance of the derive macros. `InitFromScratch` is configurable with a couple helper attributes (docs about the macro can be found [here](https://docs.rs/mdnt-support-macros/latest/mdnt_support_macros/#derive-macros)).